cpp/include/cuml/experimental/fil/detail/infer/gpu.cuh

```diff
@@ -150,56 +157,89 @@ std::enable_if_t<D==raft_proto::device_type::gpu, void> infer(
     get_max_threads_per_sm(device) / threads_per_block
   );
 
-  // Compute shared memory usage based on minimum or specified
-  // rows_per_block_iteration
-  auto rows_per_block_iteration = specified_chunk_size.value_or(
-    index_type{1}
-  );
-  auto constexpr const output_item_bytes = index_type(sizeof(
-    typename forest_t::io_type
-  ));
-  auto output_workspace_size = compute_output_size(
-    row_output_size, threads_per_block, rows_per_block_iteration
-  );
-  auto output_workspace_size_bytes = output_item_bytes * output_workspace_size;
-  if (output_workspace_size_bytes > max_shared_mem_per_block) {
-    throw unusable_model_exception(
-      "Model output size exceeds available shared memory"
+  index_type shared_mem_per_block{};
```

Comment:
Since we're already using [AAA style](https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/) everywhere in the new codebase and it is pretty useful for managing the heavy templating we're doing, let's stick to it here as well.